controller/customdomain_controller.go

```diff
@@ -281,6 +280,17 @@ func (r *CustomDomainReconciler) Reconcile(ctx context.Context, request ctrl.Req
 					},
 				},
 			}
+			if instance.Spec.RouteSelector != nil {
+				customIngress.Spec.RouteSelector = &metav1.LabelSelector{
```

Comment:
Can we just perform a deepcopy here instead of picking out `MatchSelector`? I'm concerned we are missing everything from the CR. Same with NamespaceSelector.